Survival Analysis

cheapsteak opened the pull request:
![image](https://cloud.githubusercontent.com/assets/743976/16570158/57b00b32-420e-11e6-949d-1ae680325626.png)

cheapsteak committed:
Refactor: analyses strings. Add "survival analysis"

cheapsteak committed:
Refactor: abstracts analysis launching ops into _launchAnalysis

cheapsteak committed:
add Survival Analysis set launcher

cheapsteak committed:
Stub in Survival Analysis filse

Commit:
boilerplate for survival analysis.

cheapsteak committed:
minor: bugfix: style: loading animation alignment

cheapsteak committed:
Start of survival graph, there is a graph

Commit:
Basic API for Survival Analysis is now working.

cheapsteak committed:
use 1.5 component

Commit:
PR Feedback.

Commit:
Cleaned conditional for censured/dead donors.

Commit:
Fixes lombok compile issue with importing static builder.

cheapsteak committed:
Fix jshint errors

Commit:
Uses union analyzer to exclude donors from intersection of sets for survival

Commit:
Cleans up analyzer and filters for donors that have survival data

cheapsteak committed:
Hookup graph to backend, fix graph orientation

Commit:
PR Feedback.

cheapsteak committed:
delete mock data. process donors

cheapsteak committed:
Add censored donor markers

cheapsteak committed:
Have censor markers use same color as lines

cheapsteak committed:
Standardize event parameters (event first). User shouldn't have to know about d3

cheapsteak committed:
Have a single donor list (don't groupBy status). Set status on svg element as attribute

cheapsteak committed:
style survival graph points in css

cheapsteak committed:
minor copy changes for survival plot

cheapsteak committed:
Survival Labels and click to open donor in new tab

cheapsteak committed:
minor: spacing

cheapsteak committed:
Refactor: denoramlize donors, intervals aren't necessary for graph generation

cheapsteak committed:
minor: Remove unused processIntervals function

cheapsteak committed:
Extract logic for fetching data away from component. Rename compoent to add 'graph'

cheapsteak committed:
Add survival analysis table for overall survival

cheapsteak committed:
bugfix: clicking on point would not open window

due to point shrinking when :hover changes to :active

cheapsteak committed:
Code cleanup

cheapsteak committed:
Minor: make jshint happy

cheapsteak committed:
pass palette in as input prop

cheapsteak committed:
Fetch set meta data and merge with survival data.

cheapsteak committed:
Add Legend and allow toggling datasets

Commit:
Added basic support for disease free survival

btiernay commented on dcc-portal-ui/app/scripts/survivalanalysis/js/survival-analysis.js:
```diff
@@ -0,0 +1,302 @@
+        .attr('transform', 'translate(' + margin.left + ', ' + margin.top + ')');
+      var setColor = palette[i % palette.length];
+
+      // draw the data as an svg path
```
Please capitalize first word of comments.

btiernay commented on dcc-portal-ui/app/scripts/survivalanalysis/views/survival-analysis.html:
```diff
@@ -0,0 +1,31 @@
+
+  <script type="text/template" class="survival-tip-template">
+    <div>
+      <strong><%= donor.id %></strong>
```
What kind of template is this?

cheapsteak replied:
lodash template, I couldn't figure out how to get this working with angular templates, kept trying to compile `{{}}`s

btiernay commented on dcc-portal-ui/app/styles/styles.scss:
```diff
@@ -411,6 +412,9 @@ a.no-style {
   }
   i {
     height: 1em;
+    display: flex;
```
IIRC, We've had issues with flex on some browsers. How widely supported is this?

cheapsteak replied:
IE10+

Previous issues might have been due to not using vendor prefixes to support IE10 and old safari. That is now taken care of by Autoprefixer

btiernay commented on dcc-portal-ui/app/scripts/survivalanalysis/js/survival-analysis.js:
```diff
@@ -0,0 +1,302 @@
+          .then(function (responses) {
+            return processResponses(responses);
+          })
+        ;
```
This normal formating?

cheapsteak replied:
I'm not attached to this but it makes continuations easier to work with
Would prefer semi-colon-free for continuations to be honest :P

btiernay replied:
Ok, just checking!

btiernay commented on dcc-portal-ui/app/scripts/survivalanalysis/js/survival-analysis.js:
```diff
@@ -0,0 +1,302 @@
+
+      function fetchOverallSurvival (setIds) {
+        var data = setIds;
+        
```
Why 2 lines?

cheapsteak replied:
Must have been from refactoring
btw eslint can catch these kind of things automatically, _nudge nudge_

btiernay replied:
Save these great suggestions for our build refactor. Hopefully we can slide this in after Survival Analysis.

btiernay commented on dcc-portal-ui/app/scripts/survivalanalysis/js/survival-analysis.js:
```diff
@@ -0,0 +1,302 @@
+    // draw y axis
+    svg.append('g')
+      .attr('class', 'y axis')
+      .attr('transform', 'translate(' + margin.left + ', ' + margin.top + ')')
```
Its very common to create a wrapper for `translate` in a lot of the d3 code I've seen. Cuts down on a lot of boilerplate string concat'

btiernay commented on dcc-portal-ui/app/scripts/survivalanalysis/js/survival-analysis.js:
```diff
@@ -0,0 +1,302 @@
+
+  var module = angular.module('icgc.survival', ['icgc.donors.models']);
+
+  function noop () {}
```
https://lodash.com/docs#noop

btiernay commented on dcc-portal-ui/app/scripts/survivalanalysis/js/survival-analysis.js:
```diff
@@ -0,0 +1,302 @@
+      this.$onDestroy = function () {
+        window.removeEventListener('resize', update);
+      };
+
```
Why sometimes blank line, others not?

cheapsteak replied:
Blank line between function definitions for ease of scanning, is that not typically done here?
We can make an eslinting rule for consistency's sake

btiernay replied:
No, blank lines _between functions_ makes a lot of sense.

cheapsteak replied:
Also between continuations

btiernay commented on dcc-portal-ui/app/scripts/analysis/js/analysis.js:
```diff
@@ -151,6 +151,18 @@
+      // ----- START SURVIVAL MOCK
+      
+      if ($scope.analysisType === 'survival') {
+        // $scope.analysisResult = null;
```
This needed? If not, is it safe to remove?

btiernay commented on dcc-portal-ui/app/scripts/analysis/js/analysis.js:
```diff
@@ -200,6 +212,34 @@
+      },
+      survival: {
+        name: 'Survival Analysis',
+        description: 'TBD',
```
Maybe we should nail this down?

cheapsteak replied:
This section _might_ get removed as I don't know if was intended to be a standalone experiment type

btiernay replied:
What about a description. Do you have copy for that? Perhaps we can steal something from the spec in the meantime?

cheapsteak replied:
Sorry just saw

Here is the description from the wiki

> Display the survival analysis of your donor sets and compare some characteristics such as gender, vital status and age at diagnosis between you donor sets.

btiernay commented on dcc-portal-ui/app/scripts/survivalanalysis/js/survival-analysis.js:
```diff
@@ -0,0 +1,302 @@
+      var ctrl = this;
+      var graphContainer = $element.find('.survival-graph').get(0);
+      var svg = d3.select(graphContainer).append('svg');
+      var tipTemplate = _.template($element.find('.survival-tip-template').html());
```
Ok, I guess this is the template. Why not just use an angular template?

cheapsteak replied:
I couldn't figure out how to get the parent controller to stop trying to compile `{{}}` bindings
Once we switch to webpack the template can be in its own html file and required in (instead of living in another template's script tag)

btiernay replied:
Maybe you could use a directive?

cheapsteak replied:
That should work, seems a bit heavy for 8 lines of template though 
Wish I could just have this in a separate file and require it in

btiernay replied:
Technically all of this should probably go in a directive since it dom manipulation heavy. If you did that, you would have a natural place to hook into the directive template. It seems as though this has been a common pattern in our code base at least.

btiernay commented on dcc-portal-ui/app/scripts/survivalanalysis/js/survival-analysis.js:
```diff
@@ -0,0 +1,302 @@
+            $scope.$emit('tooltip::hide');
+          },
+          onClickDonor: function (e, donor) {
+            window.open('/donors/'+donor.id, '_blank');
```
Are you sure this is what we want to do? Usually we navigate to the page using the current window via location service (I think)

cheapsteak replied:
I'll check

cheapsteak replied:
Junjun says new tab, easy to change though :)

btiernay commented on dcc-portal-ui/app/scripts/survivalanalysis/js/survival-analysis.js:
```diff
@@ -0,0 +1,302 @@
+
+  module
+    .service('SurvivalAnalysisService', [
+      '$q',
```
Is this needed? We are using ngAnnotate.

cheapsteak replied:
Was not aware that this is no longer necessary, will refactor

btiernay replied:
Np. Don't quote me on that, but I'm pretty sure. Might want to double check the grunt file first.

btiernay commented on dcc-portal-api/src/main/java/org/icgc/dcc/portal/analysis/SurvivalAnalyzer.java:
```diff
@@ -0,0 +1,201 @@
+    currentInterval.setCumulativeSurvival(cumulativeSurvival);
+
+    for (int i = 0; i < time.length; i++) {
+
```
Remove line

btiernay commented on dcc-portal-api/src/main/java/org/icgc/dcc/portal/analysis/SurvivalAnalyzer.java:
```diff
@@ -0,0 +1,201 @@
+    val entitySetMapBuilder = new ImmutableMap.Builder<UUID, UnionUnit>();
+
+    for (val id : sets) {
+      val exclusions = sets.stream()
```
This seems very strange to stream over and over in the loop. This entire method could probably be rewritten to not have the loop.

btiernay replied:
Actually, it's probably fine the way it is. I would suggest a more consistent naming scheme for the variables:

``` java
for (val setId : setIds) {...
```

btiernay replied:
Heh, maybe this would just work:

``` java
import static Sets.difference;
import static Maps.uniqueIndex;
import static Collections.singleton;
...
return uniqueIndex(setIds, setId -> new UnionUnit(singleton(setId), difference(setIds, singleton(setId)));
```

Commit:
Adds support for disease free data in survival analysis.

Commit:
Adds support for disease free data in survival analysis.

Commit:
Merge branch 'survival-disease-free-api' of ssh://github.com/icgc-dcc/dcc-portal into survival-disease-free-api

Commit:
Added comment for query against donor type in elasticsearch.

cheapsteak committed:
Merge pull request #8 from icgc-dcc/survival-disease-free-api

Survival disease free api

cheapsteak committed:
Minor: style changes based on suggestions

cheapsteak committed:
Refactor. Add disease free survival chart to phenotype analysis

cheapsteak commented:
@btiernay please help deploy this to dev once build is done

![image](https://cloud.githubusercontent.com/assets/743976/16601217/5970fa32-42d9-11e6-98e9-a8f7b8862476.png)

btiernay commented:
@cheapsteak Done!

btiernay commented:
One thing I noticed is that clicking around on the legends of the second chart, sometimes a series doesn't come back. If you try a few permutations you will probably get this behavior.

btiernay commented:
Ah, could be because there is no data for Brain :) Nvm

btiernay commented:
How hard would it be to add a checkbox beside each of the series in the legend? That way, a user would know if they could enable / disable them. You could use the web fonts for this like we do in the facets.

btiernay commented:
Also, the dynamic scaling of the axis labels can lead to the text being difficult to read. Anyway we can keep the font size constant?

btiernay commented:
Oh, btw, looks good!

cheapsteak commented:
Thanks!

Checkboxes are a good idea. Not hard, 15-20 minutes

Are font sizes not currently constant? The reason I'm handling resizing the way I am now (redrawing the svg instead of just updating the width and height attributes on the tag) was so that font and ticks would be appropriate sizes, but I haven't confirmed that yet. 

Can we open up dev to externdal IPs but put it behind basic auth? I've got a macbook pro and a pc at home, would be convenient to be able to check on it here without having to haul $3k worth of equipment back and forth on bike :P

btiernay commented:
That might be a no go, but there is a way. Will give instructions in a bit.

btiernay commented:
Looking at the other titles, should it be `Survival analysis`? I think this a French capitalization scheme :)

cheapsteak committed:
remove survival analsyis from data analysis launcher

cheapsteak committed:
Remove initially disabled data prop

cheapsteak committed:
Bugfix: colors of sets in table don't match colors in graph

scope.item.entitySetIds don't follow the order

cheapsteak committed:
Bugfix: update tip to reflect survivalTime -> time rename

Commit:
Relaxes filtering for disease free.

cheapsteak committed:
Bugfix: data in table showing up in wrong order

cheapsteak committed:
Add Bootstrap Grids back. adds 9kb minified, 1kb min & gzipped

cheapsteak committed:
Do not track compiled bootstrap

cheapsteak committed:
Bring graphs side by side using bootstrap grid

cheapsteak committed:
Add option to use line/ticks instead of circles for survival markers

cheapsteak committed:
Hide deceased markers from survival graph

cheapsteak committed:
Fix autoprefixing task. Upgrade to postcss, remove deprecated grunt-autoprefixer

cheapsteak committed:
Add checkbox to legend. Use correct cursor for non-interactable sets

btiernay commented:
Conflicts!

cheapsteak committed:
Merge branch 'develop' of github.com:icgc-dcc/dcc-portal into feature/survival-plot

# Conflicts:
#	dcc-portal-ui/app/scripts/analysis/views/analysis.analysis.launcher.html

Commit:
Fix for how survival plot donors are filtered and how the intervals are generated.

 Should produce much more data now.

Commit:
Merge branch 'feature/survival-plot' of ssh://github.com/icgc-dcc/dcc-portal into feature/survival-plot

cheapsteak committed:
Refactor: SetService can create entitysets without navigating away from the page

cheapsteak committed:
Wrap `createEntitySet` with `Page.startWork`. remove createAndViewEntitySet

The filters created for createAndViewEntitySet only works for going to the 'repositories' page, not for /search

cheapsteak committed:
Clicking on Survival Analysis donor numbers will go to advanced search with results

cheapsteak committed:
Minor: capitalization

cheapsteak committed:
minor: newline each dependency, line too long

cheapsteak committed:
remove link to nonexistant bower's invariant/browser.js

cheapsteak committed:
Add feature to export tsv

cheapsteak commented:
![image](https://cloud.githubusercontent.com/assets/743976/16696720/272ee3da-4514-11e6-8c9a-0b1da39bcbe0.png)

Can now download tsv files

cheapsteak committed:
Set operations now visible next to survival table. clickable

cheapsteak committed:
Fix jshint warnings

cheapsteak commented:
Set venn preview now available 

![image](https://cloud.githubusercontent.com/assets/743976/16701568/96073978-452e-11e6-806b-52fc3550fa63.png)

Click to go to set page

btiernay commented:
As mentioned, would be nice to:
- Move legends to under graph x-axis as to not overlap data
- Outline / fill the sets with the series colour

How much effort would each of these items be?

cheapsteak committed:
Respond to github comments

cheapsteak committed:
Move survival graph legend to the bottom

cheapsteak committed:
Only show venn diagram if intersections exist

cheapsteak committed:
Jshint: missing semicolon

andricDu commented:
retest this please

btiernay commented:
Ok to test

cheapsteak commented:
retest this please

Commit:
SurvivalAnalysis now has a property to notify if intersection is present.

cheapsteak committed:
Merge branch 'develop' of github.com:icgc-dcc/dcc-portal into feature/survival-plot

cheapsteak committed:
Fix grunt connect:dist

cheapsteak committed:
Update ng-annotate

cheapsteak committed:
Styling: remove padding on survival graph table

cheapsteak committed:
DCC-4985 Update launch copy to include survival. alignment

cheapsteak committed:
Bugfix: venn not showing up when it should

cheapsteak committed:
Add .t_tabs—smaller for smaller tabs

cheapsteak committed:
Add class .tabs--no-accent

cheapsteak committed:
Move survival graphs into tabs

cheapsteak committed:
Initialize activeSurvivalGraph to `overall`

cheapsteak committed:
Add export button, with tooltip

cheapsteak committed:
Added tooltip to explain donors included in the analysis DCC-5011

cheapsteak committed:
Merge branch 'develop' of github.com:icgc-dcc/dcc-portal into feature/survival-plot

# Conflicts:
#	dcc-portal-api/src/main/java/org/icgc/dcc/portal/repository/TermsLookupRepository.java

cheapsteak commented:
Tabs for the two graphs
Tooltips

![image](https://cloud.githubusercontent.com/assets/743976/16845257/48636b10-49b6-11e6-9829-9cfe3d551f77.png)

Waiting for build

cheapsteak committed:
Embiggen export icon and shift it up
